src/npctalk.cpp

```diff
@@ -317,14 +321,47 @@ static void tell_veh_stop_following()
 
 static void assign_veh_to_follow()
 {
-    for( auto &veh : g->m.get_vehicles() ) {
-        auto &v = veh.v;
+    for( wrapped_vehicle &veh : g->m.get_vehicles() ) {
+        vehicle *&v = veh.v;
         if( v->has_engine_type( fuel_type_animal, false ) && v->is_owned_by( g->u ) ) {
             v->activate_animal_follow();
         }
     }
 }
 
+static void tell_magic_veh_to_follow()
+{
+    for( wrapped_vehicle &veh : g->m.get_vehicles() ) {
+        vehicle *&v = veh.v;
```

Comment:
```suggestion
        vehicle *v = veh.v;
```
Or
```suggestion
        vehicle &v = *veh.v;
```
But a *reference* to a pointer is unnecessary. The pointer stored `veh.v` isn't going to change during the loop body, so you might as well store a copy of that pointer as local variable.